File: bdp-master-server/src/main/java/org/bdp/master/server/service/ServerService.java
package org.bdp.master.server.service;

import org.bdp.master.server.domain.Server;

import java.util.List;

public interface ServerService {

    Server findOne(Long id);

    Server findOne(String hostname);

    List<Server> findAll();

    void save(Server server);

    void delete(Long id);

    void loadAll();

}

File: bdp-master-client/src/main/scala/org/bdp/master/client/service/ServerService.scala
package org.bdp.master.client.service

import com.typesafe.scalalogging.LazyLogging
import org.bdp.master.client.domain.Server
import org.bdp.master.client.util.{JsonDecoder, RedisClient}
import org.bdp.master.client.Constants._

object ServerService extends LazyLogging{

  def getServerBy(id: Long): Server = {
    JsonDecoder.decodeServer(RedisClient.get(s"$SERVER_KEYSPACE:$id"))
  }

  def getServerBy(hostname: String): Server = {
    val key = RedisClient.get(s"$INDEX_PREFIX$SERVER_KEYSPACE:$hostname")
    JsonDecoder.decodeServer(RedisClient.get(key))
  }

}
